control/available_plugin.go

```diff
@@ -457,6 +458,10 @@ func (ap *availablePlugins) publishMetrics(contentType string, content []byte, p
 		return []error{serror.New(ErrPoolNotFound, map[string]interface{}{"pool-key": key})}
 	}
 
+	if !pool.HasSubscription(taskID) {
+		return []error{serror.New(ErrBadTaskId, map[string]interface{}{"task-id": taskID})}
+	}
+
```

Comment:
@IRCody after thinking about I can't find any scenario which would lead apart from artificially pass incorrect task id in unit tests. I will remove the commit  which introduced it